openapi3/schema.go

```diff
@@ -837,6 +838,35 @@ func (schema *Schema) visitSetOperations(settings *schemaValidationSettings, val
 	}
 
 	if v := schema.OneOf; len(v) > 0 {
+
+		if schema.Discriminator != nil {
+			/* Find mapped object by ref */
+			if valuemap, okcheck := value.(map[string]interface{}); okcheck {
+				pn := schema.Discriminator.PropertyName
+				if discriminatorVal, okcheck := valuemap[pn]; okcheck {
+					if len(schema.Discriminator.Mapping) > 0 {
+						if mapref, okcheck := schema.Discriminator.Mapping[discriminatorVal.(string)]; okcheck {
```

Comment:
Can't casting to string go away? There's a cast to `map[string]interface{}` before.

Reply:
Do you mean to to type assert to `map[string]string` (<- this doesn't work)

Reply:
erlang would have been easier I guess :)

Reply:
I meant the casting on this line i.e. `discriminatorVal.(string)`

Reply:
If I don't force the cast I get this error
`schema.go:848:57: cannot use discriminatorVal (type interface {}) as type string in map index: need type assertion`

Reply:
Oh right! Sorry, my mind stopped working...